contrib/platform/src/com/sun/jna/platform/win32/Winsvc.java

```diff
@@ -859,6 +860,8 @@ public ENUM_SERVICE_STATUS() {
      */
     @FieldOrder({"lpServiceName", "lpDisplayName", "ServiceStatusProcess"})
     public static class ENUM_SERVICE_STATUS_PROCESS extends Structure {
+        public static final TypeMapper TYPE_MAPPER = W32APITypeMapper.DEFAULT;
```

Comment:
There are statics, that are read to find option overrides, but TYPE_MAPPER is not one of them. At least I did not find that path. This fix is a no-op, as you rightly pointed out, that you can use the constructor to set the right typemapper and that is done in line 885/888.

As there is exactly one Constructor, all paths are covered.

Reply:
Are you sure? What about this line then?
https://github.com/java-native-access/jna/blob/d8ce23e2d0baf7243cf07639731c4de6ca62e412/src/com/sun/jna/Native.java#L763

I guess in this case you are right, the constructor is covered, but if there were to be another constructor that took a Pointer (as such exists in the base), then it's error prone to rely on the constructor only.

Reply:
Ah - thanks for the pointer, I overlooked that. As it works correctly I prefer not to fix it (there are much more things that are error prone when interfacing with native, this is one of the lesser concerns). In fact, should someone decide to make Winsvc implement Library it would again invalidate this fix.

https://github.com/java-native-access/jna/blob/d8ce23e2d0baf7243cf07639731c4de6ca62e412/src/com/sun/jna/Native.java#L735-L740

At this time we hit 739 and thus get `ENUM_SERVICE_STATUS_PROCESS` itself.